### PR TITLE
Update xgboost user guide

### PR DESCRIPTION
**Pull request by SYangster**
### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: docs/user_guide/federated_xgboost/secure_xgboost_user_guide.rst [hunk 1]
```diff
@@ -20,24 +20,24 @@ It supports federated training in the following 4 modes:
...
-    pip install nvflare~=2.4.2
+    pip install nvflare~=2.5.1
 
-A version of XGBoost that supports secure mode is required, which can be installed using this command,
+It requires XGBoost 2.2 or above, which can be installed using this command,
 
 .. code-block:: bash
 
-	pip install https://s3-us-west-2.amazonaws.com/xgboost-nightly-builds/vertical-federated-learning/xgboost-2.1.0.dev0%2Bde4013fc733648dfe5c2c803a13e2782056e00a2-py3-none-manylinux_2_28_x86_64.whl
+	pip install xgboost
```
Comment (YuanTingHsieh) [inline docs/user_guide/federated_xgboost/secure_xgboost_user_guide.rst:40]: `pip install "xgboost>=2.2"`?

File: examples/advanced/xgboost/histogram-based/README.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ tensorboard --logdir /tmp/nvflare/xgboost_v2_workspace/simulate_job/tb_events
 
 ## Timeout configuration
 
-Please refer to [Federated XGBoost Timeout Mechanism](https://nvflare.readthedocs.io/en/2.4/user_guide/federated_xgboost/timeout.html)
+Please refer to [Federated XGBoost Timeout Mechanism](https://nvflare.readthedocs.io/en/2.4/user_guide/federated_xgboost/reliable_xgboost_timeout.html)
```
Comment (YuanTingHsieh) [inline examples/advanced/xgboost/histogram-based/README.md:36]: link should change 2.4 -> main? we could prob. move these link update in another PR.

**Review by YuanTingHsieh — COMMENTED** [review]
Mostly LGTM

File: examples/advanced/xgboost/histogram-based/README.md (hunk 2 above)
Comment (SYangster) [reply examples/advanced/xgboost/histogram-based/README.md:36]: Right I can update the links once we move the 2.4 work to main. The link change here is needed to pass the premerge check.

Comment (YuanTingHsieh) [reply examples/advanced/xgboost/histogram-based/README.md:36]: Ah I did not notice this PR is against 2.4

**Review by YuanTingHsieh — APPROVED** [review]
LGTM
